tools/scan-build-py/libscanbuild/analyze.py

```diff
@@ -165,22 +165,28 @@ def generate_func_map_lines(fnmap_dir):
                 for line in in_file:
                     yield line
 
-    def write_global_map(ctudir, mangled_ast_pairs):
+    def write_global_map(ctudir, arch, mangled_ast_pairs):
         """ Write (mangled function name, ast file) pairs into final file. """
 
-        extern_fns_map_file = os.path.join(ctudir, CTU_FUNCTION_MAP_FILENAME)
+        extern_fns_map_file = os.path.join(ctudir, arch,
+                                           CTU_FUNCTION_MAP_FILENAME)
         with open(extern_fns_map_file, 'w') as out_file:
             for mangled_name, ast_file in mangled_ast_pairs:
                 out_file.write('%s %s\n' % (mangled_name, ast_file))
 
-    fnmap_dir = os.path.join(ctudir, CTU_TEMP_FNMAP_FOLDER)
+    triple_arches = glob.glob(os.path.join(ctudir, '*'))
```

Comment:
Is this the idiomatic way to enumerate the subdirectories of a directory?

Reply:
I'm afraid yes.